source/includes/usage-examples/code-snippets/Watch.java

```diff
@@ -0,0 +1,34 @@
+package usage.examples;
+
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import com.mongodb.client.ChangeStreamIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.changestream.FullDocument;
+
+public class Watch {
+    public static void main( String[] args ) {
+
+        // Replace the uri string with your MongoDB deployment's connection string
+        String uri = "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
+
+        try (MongoClient mongoClient = MongoClients.create(uri)) {
+
+            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
+            MongoCollection<Document> collection = database.getCollection("movies");
+
+            List<Bson> pipeline = Arrays.asList(
+                Aggregates.match(
+                        Filters.in("operationType",
+                                Arrays.asList("insert", "update"))));
+            ChangeStreamIterable<Document> changeStream = database.watch(pipeline)
+                .fullDocument(FullDocument.UPDATE_LOOKUP);
+            changeStream.forEach(event -> System.out.println("Received a change to the collection: " + event));
```

Comment:
The only downside to this approach is there's no clear way to stop / exit the process.  Which will lead to questions similar to: https://developer.mongodb.com/community/forums/t/how-to-unwatch-or-stop-a-running-change-stream-with-mongodb-java-driver/6760

I don't have a clear answer other than also suggesting documenting something like:

```
try (MongoChangeStreamCursor<Document> cursor = collection.find().cursor()) {
     while (cursor.hasNext()) {
         System.out.println("Received a change to the collection: " + cursor.next());

         if (shutdown()) {
            break;
         }
     }
 }
```

Reply:
Unfortunately I feel like `forEach()` is probably the most useful function to demonstrate for our target audience of inexperienced developers -- next/hasNext ends up being a lot of verbosity to accomplish the same thing. What if I added some details about the cursor remaining open instead:

```

    … the cursor remains open until one of the following occurs:

        The cursor is explicitly closed.
        An invalidate event occurs.
        If the deployment is a sharded cluster, a shard removal may cause an open change stream cursor to close, and the closed change stream cursor may not be fully resumable.
```
... so even if it's not clear how to close the stream from code, the circumstances under which it will close are made clear?

I could also add an explicit cursor close in the example if you think that would be helpful. But I wonder if we're making it more complicated than it needs to be.

Reply:
@ccho-mongodb what do you think? While I do want to be comprehensive and cover closing a stream, I worry that it might just overcomplicate the examples which are supposed to be straightforward.

Reply:
Some background info that I think we've already discussed, but just want to preface my recommendation:
In the [Node.js version of this usage example](https://docs-mongodbcom-staging.corp.mongodb.com/da1c19e/node/docsworker-xlarge/preview-071020/usage-examples/changeStream), we had the same app setup the callback, have it exit after it receives the expected event, and then used timers to ensure it exited regardless of success. In hindsight, I wonder if this example is trying to strictly adhere to the documentation template structure we set for Usage Examples, and perhaps we should reduce this example to setup and iteration, and link to a more complete example to Fundamentals that walks them through events that trigger the callbacks to keep the code clean and more reflective of how you might use them.

So my order of preference without any user study input:
1) Only show code to watch a pipeline and iterate through ChangeStream event and link to Fundamentals where we walk through sending events as well.
2) Add code to exit after N events received, explain that they need to kill the app if it does not get the expected number
3) Use a timer to exit automatically

As far as `hasNext` or `forEach`, I'd prefer the latter *only if we can accomplish what we need using it*. I don't think the inexperience/verbosity is an obstacle, but we're using Java 8 try-with-resources, so we should be able to assume they are also familiar with forEach.

Reply:
I think demonstrating callbacks actually being called/sending events is pretty crucial to understanding how change streams work, otherwise you miss out on the entire point -- changes! I think code to exit after N events is an OK solution to make sure the example exits, but I prefer a timer. I'll try both and see if I can come up with something that doesn't feel too clunky.

Reply:
Ok, I know that you already know this, but keep in mind that content in the Usage Examples is for showing working code examples (generally based on search results) and providing a quick reference to them. The content in Fundamentals are for understanding how to use different methods to accomplish a specific task. There's been some overlap, but it would be great to keep this separation so there's an expected pattern of visiting Fundamentals for further learning if necessary, and that we're not repeating ourselves in the content too much.

Reply:
The current form is a working example that provides a quick reference. Does that mean we should leave this as is?